### DISCUSSION TITLE
Very nice to see this happening!

### DISCUSSION THREAD

**Discussion by DJMaikie** [geodesicsolutions-community/geocore-community#155] [General]
Hi there, 

Not sure if this is the right place for this message but just to let you know i am very interested and eager to use your new product.

As a still active user of the late geodesic.com script i am very pleased to see your initiative guys!
In fact I was just looking around if i needed to change products on my production site but rather stay with the geodesic script.

Could you let me know if the beta 3 version is safe to be used on a production site?

Keep up this great Job!

Cheers Maikie

**Comment by vicos59** [comment]
Hi Maikie,

Not sure I would use Beta-3 in production (not on  PHP 8+ for sure). Certainly do not use Beta-4 since there are a lot of bugs due to upgrades of major libraries.

IMO, Beta-5 should be pretty solid. @jonyo is the primary developer and he could best say when he is ready to release that. I'm just his helper.

Actually, if you wanted to install a copy on your own server for in-house testing and don't want to wait until Beta-5, you could use the current codebase, if you know how to use composer to build a release. See #153. As a matter of fact, I intend on doing that very thing tomorrow so I can begin the process of updating my old templates.

There are only a few small items outstanding in the current copy which you can see in #95 

Hope this helps. 

P.S. Be sure to follow the install/upgrade docs at https://geodesicsolutions.org/wiki ... Several extra steps you need to take, mainly due to removing the licensing and removing an obsolete addon (AdPlotter).

**Reply by DJMaikie** [reply]
Thanks Vicos, 

I am still on PHP 7 so will not update / move to php 8 and the CE edition of Geocore until it is safe to do so.  I am considering to move to Beta but perhaps i am better of waiting for Beta 5 i guess? And to be sure: The payment gateways will still work though? (i Use iDeal) 

cheers Maikie

**Reply by jonyo** [reply]
With @vicos59 's help we are finding bugs that have been around for a long time, I guess places that are not used often or maybe the bug is not too big.  So I think the next beta 5 will be very stable, perhaps even more than the versions before (since we are finding and fixing bugs that were present on the older 18.x, 17.x and even older versions).  Especially if you are on PHP 7.4.  Now most of the bugs we are tackling are related to PHP 8 compatibility.   Plus some lingering issues related to the Smarty template engine update.

And just to make sure it is clear, the software will continue to work with PHP 7.4.  We've tried looking at earlier PHP 7 version compatibility but I don't think we have the ~man~ developer-power to also keep it compatible with "end of life" PHP versions.

I don't know when beta 5 will be out.  At the very least not until all the main issues reported in #95 are resolved.  Also just to help manage expectations, at the moment I'm taking a breather from working on it so I don't burn out 😆 .  And right now I'm the main software engineer actively working on it, so that means you'll see a pause in progress.  Though @vicos59 has been doing some work too, he's been a great help!  Even with making changes in the software.  I know starting out it can be intimidating, if you have not done much "public work" it may seem like helping out is out of your league.  But the things that need doing, some of them anyone can do, not code related at all.  Like trying it out and reporting problems.  Or even just contributing to discussions here in the discussion section!  That is actually a huge help as well, one 🧠 can never be as "smart" as a bunch of 🧠 working together.  But I digress...

**Reply by jonyo** [reply]
> And to be sure: The payment gateways will still work though? (i Use iDeal)

I meant to answer this as well:  I can't say for sure, I do not know if @vicos59 has specifically tested the payment gateways beyond the main one he uses on his site, Paypal Pro.

What version are you currently using?  If it is a recent-ish one (like maybe 16 or up), and the gateway works for you now, I think it would work on the new version.  I checked and the main file used for that payment gateway has not had changes since 2015.  Though that is no guarantee, there is the chance that changes to the main system broke an individual payment gateway... But personally I have a feeling it would work if you see it was working still in newer releases.

But that does bring up a good point, we need to be sure to test all the different payment gateway options somehow.

**Comment by vicos59** [comment]
> But that does bring up a good point, we need to be sure to test all the different payment gateway options somehow.

I'll only be testing PayPal and Paypal Pro.

IMO it would take a lot of time to test all of those gateways just for the fact that one would have to make their way to the respective developer sites for the gateway, research if they have a test/sandbox setup available, register and implement on the test system and execute some payment tests for each gateway.

Since this is a community project, my feeling is that if someone needs a particular gateway verified, they should install a test copy on their system and if it doesn't work, come back here and open an Issue. There is only so much you can expect volunteers to do (speaking of burning out) and the people that benefit need to step up to the plate where possible. I hope that doesn't sound too unreasonable. BTW, I'm speaking in general and not trying to single out this particular inquiry.

**Reply by DJMaikie** [reply]
I fully understand that you guys are not paid and appreciate your efforts. Also expectations are clear on having no timetables so no pressure there :-) 

I am not a coder so i am afraid i can't assist you guys here. :-( But I am more than happy to help if and where i can eg. test things and report possible bugs . FYI: I am on 18.02 now and all works fine so far (apart from some minor template lay-out "flaws") 

You suggest docker as local server? Or should i use my current WAMP/XAMP for this? Also i am thinking of building a totally new and clean site in a different folder on my current VPS with the CE version. And i could begin transfer the Ads and users into that new site once i am finished setting it up properly. 

If you have suggestions in this matter they are more than are welcome :-)

Cheers Maikie

**Reply by vicos59** [reply]
From my perspective, I do 2 levels of testing:

DOCKER: This is the environment @jonyo has setup for development. So, I use it for testing as it make the process very easy. I find bugs and report them, then he fixes them. Then I can easily retest with out having to manually update my local server.

MY SERVER: When we have a new Beta release, I upgrade the installation on my test domain on the production server so I can test it in my real environment. It also allows me to work on updating my own templates w/o impacting the development environment.

I'm not sure if testing Payment Gateways will function properly on Docker since it uses localhost. I plan to test Paypal on my test domain shortly.

> Also i am thinking of building a totally new and clean site in a different folder on my current VPS with the CE version. And i could begin transfer the Ads and users into that new site once i am finished setting it up properly.

I was considering doing something similar, but haven't decided yet.

Hope this helps.

**Reply by vicos59** [reply]
P.S. I'm pretty sure that the Facebook Login Addon is not functioning properly, at least I can not get it work on my test domain. The GeoCore codebase is 2 major versions behind the FB PHP libraries (before FB abandoned them). In case this is something you require.

**Reply by vicos59** [reply]
P.S.S. If you just want to test outside of Docker, I already built a snapshot release (.zip) of the latest changes. I can upload it to one of the file sharing sites if you like. I don't have privs to do it here in Github.
